aten/src/ATen/core/CheckMemoryFormat.h

```diff
@@ -7,8 +7,7 @@ check_tensor_options_and_extract_memory_format(
     const TensorOptions& options,
     std::optional<MemoryFormat> memory_format) {
   TORCH_CHECK(
-      options.requires_grad_opt() == std::nullopt ||
-      options.requires_grad_opt().value() == false,
+      options.requires_grad_opt() != true,
```

Comment:
ngl I don't like this idiom

Reply:
You like more explicit way?

Reply:
I like when the lhs and rhs have same type, yes. This is a taste thing I suppose but knowing there's an optional involved makes me less likely to make other mistakes